tests/rados/test_bilog_trim.py

```diff
@@ -0,0 +1,143 @@
+"""
+Tier-2 test to ensure PG deep-scrub does not report inconsistent PGs
+when radosgw bilog trimming occurs on PGs where secondary OSD is down
+Customer Bug: 2056818 - [GSS][RADOS][RGW] Scrub errors (omap_digest_mismatch) on
+PGs of RGW metadata pools after upgrade to RHCS 5
+
+Test needs to run as part of rgw/tier-2_rgw_rados_multisite_ecpool.yaml
+as it needs a rgw multisite setup along with data present in RGW pools
+"""
+
+import datetime
+import random
+import time
+from math import floor
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados import utils as rados_utils
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-83575437
+    #BZ-2056818
+    This test is to verify no issues are observed during deep-scrub when rgw bilog
+    trimming occurs with OSD down
+    Steps:
+    Ref:
+    - https://bugzilla.redhat.com/show_bug.cgi?id=2056818#c42
+    - https://bugzilla.redhat.com/show_bug.cgi?id=2056818#c50
+    1. Deploy minimal RGW multisite setup
+    2. Add data to both the sites in S3 buckets and let the minimal client IO running
+    3. Keep running the deep-scrub for index pool PGs
+    4. Set noout flag
+    5. stop the secondary OSD on the primary site(site1)
+    6. run the following command
+       radosgw-admin bilog autotrim
+    7. Once the above command finishes, start the primary OSD that was stopped in step 4
+    8. Retrigger the deep-scrub for eight PGs in index pool
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+
+    try:
+        # set noout flag
+        if not rados_utils.configure_osd_flag(ceph_cluster, action="set", flag="noout"):
+            log.error("Could not set noout flag on the cluster")
+            raise Exception("Could not set noout flag on the cluster")
+
+        # get acting set for the rgw index pool
+        _pool_name = "primary.rgw.buckets.index"
+        acting_set = rados_obj.get_pg_acting_set(pool_name=_pool_name)
+        log.info(f"Acting set for pool {_pool_name}: {acting_set}")
+
+        # secondary osd of 1st PG from rgw index pool
+        second_osd = acting_set[1]
+
+        # get list on active OSDs
+        osd_list = rados_obj.get_osd_list(status="up")
+
+        # bring down 30% of total UP OSDs
+        osd_down = random.choices(osd_list, k=floor(0.3 * len(osd_list)))
```

Comment:
randomly selecting OSDs have potential for making PGs inactive. Is this something that is expected?

Reply:
As discussed offline, changed the code to bring down half of OSDs from any 2 OSD hosts